### ISSUE TITLE
Error when using ChannelTaskScheduler for internal-dispatcher in akka cluster node

### ISSUE THREAD

**Issue by SchiessMax** [akkadotnet/akka.net#5498]
**Version Information**
Version of Akka.NET? 1.4.31
Which Akka.NET Modules? Akka.Actor/Akka.Cluster

**Describe the bug**
When trying to use the ChannelTaskScheduler as akka.actor.internal-dispatcher an exception occurs (see stack trace below). I made the configuration according to https://github.com/akkadotnet/akka.net/pull/5403.
The exception does not occur when the ChannelTaskScheduler is used for the other dispatchers.
Please let me know if you need more information.

Stack trace:

```
Exception during creation
 ---> System.TypeLoadException: Error while creating actor instance of type Akka.Cluster.ClusterHeartbeatSender with 0 args: ()
 ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.InvalidOperationException: ValueFactory attempted to access the Value property of this instance.
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Akka.Actor.Internal.ActorSystemImpl.TryGetExtension[T](T& extension)
   at Akka.Actor.Internal.ActorSystemImpl.GetExtension[T]()
   at Akka.Actor.ActorSystemWithExtensions.WithExtension[T,TI](ActorSystem system)
   at Akka.Cluster.Cluster.Get(ActorSystem system)
   at Akka.Cluster.ClusterHeartbeatSender..ctor()
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
   at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at Akka.Actor.Props.ActivatorProducer.Produce()
   at Akka.Actor.Props.NewActor()
   --- End of inner exception stack trace ---
   at Akka.Actor.Props.NewActor()
   at Akka.Actor.ActorCell.CreateNewActorInstance()
   at Akka.Actor.ActorCell.<>c__DisplayClass116_0.<NewActor>b__0()
   at Akka.Actor.ActorCell.UseThreadContext(Action action)
   at Akka.Actor.ActorCell.NewActor()
   at Akka.Actor.ActorCell.Create(Exception failure)
   --- End of inner exception stack trace ---
   at Akka.Actor.ActorCell.Create(Exception failure)
   at Akka.Actor.ActorCell.SysMsgInvokeAll(EarliestFirstSystemMessageList messages, Int32 currentState)
```

**To Reproduce**
1. Create ActorSystem with the following hocon configuration and System Name "MyActorSystem":
```
akka {
    # Copied from default config: The ChannelTaskScheduler expects the settings in "akka.channel-scheduler", but the default is in "akka.actor.channel-scheduler"
    channel-scheduler : {
        parallelism-min : 4
        parallelism-factor : 1
        parallelism-max : 64
        work-max : 10
        work-interval : 500
        work-step : 2
    }
    actor {
        provider = cluster
        default-dispatcher = {
            executor = "channel-executor"
            channel-executor.priority = "normal"
        }
       # This is the part that makes the actor system crash on startup
        internal-dispatcher = {
            executor = "channel-executor"
            channel-executor.priority = "high"
        }
    }
    remote {
        dot-netty.tcp {
            port = 5510
            hostname = localhost
        }
        default-remote-dispatcher {
            executor = "channel-executor"
            channel-executor.priority = "high"
        }
        backoff-remote-dispatcher {
            executor = "channel-executor"
            channel-executor.priority = "low"
        }
    }
    cluster {
        seed-nodes = [
           "akka.tcp://MyActorSystem@localhost:5510"
        ]
    }
}
```
2. The error is logged on the console
3. Comment out the internal-dispatcher configuration
4. Start the ActorSystem with the updated configuration
5. No error occurs

_Links to working reproductions on Github / Gitlab are very much appreciated_

**Expected behavior**
The ActorSystem starts up correctly

**Actual behavior**
The ActorSystem did not start up correctly and logged an error.

**Screenshots**

**Environment**
Are you running on Linux? Windows? Docker? Which version of .NET?
Running on windows, but same problem in my linux VM. Dotnet version: netcoreapp3.1

**Additional context**

**Comment by Aaronontheweb** [comment]
cc @Zetanova

**Comment by Zetanova** [comment]
@Aaronontheweb  I fixed it already and then somebody reverted the PR.

The Problem is in the Cluster constructor and the use of await + Task.Result.
The thread gets free and the normal TaskPoolManager will reuse it and some other component 
that uses `Cluster.Get(sys)` will throw because of the use of Lazy locking in the Extensions list.

**Comment by Zetanova** [comment]
I have 4 ideas to resolve it
1) remove the locking of the Cluster constructor, I made already some PR to prepare for it
2) really Lock thread in the cluster constructor
3) Add an special synchronization context for the ActorSystem startup
4) Change the system.Extensions dictionary from `<Type, Lazy<IExtionsion>>` to `<Type, Task<IExtionsion>>`

**Comment by Aaronontheweb** [comment]
Resolved via #5861
